klarna-on-demand/src/main/java/com/klarna/ondemand/CryptoImpl.java

```diff
@@ -0,0 +1,116 @@
+package com.klarna.ondemand;
+
+import android.content.*;
+import android.content.Context;
+import android.util.Base64;
+
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+class CryptoImpl implements Crypto {
```

Comment:
This is not a very good class name. This is obviously an implementation of the `Crypto` interface, but what kind?